mods/translucent-windows.wh.cpp

```diff
@@ -1024,25 +1945,39 @@ void Wh_ModAfterInit()
         ApplyForExistingWindows();
 }
 
-void Wh_ModUninit(void) 
+void Wh_ModBeforeUninit()
 {
+    {
+        std::lock_guard<std::mutex> guard(rainbowWindowsMutex);
+        for (auto& [hWnd, pair] : g_rainbowWindows)
+            PostMessageW(hWnd, msgRainbowTimer, RAINBOW_UNLOAD, 0);
+    }
+}
+
+void Wh_ModUninit(void) 
+{ 
     g_settings.Unload = TRUE;
-    ApplyForExistingWindows();
+    g_drawTextState.clear();
+    g_rainbowWindows.clear();
+    RevertSysColors();
 
     std::unordered_set<HWND> subclassedWindows;
     {
         std::lock_guard<std::mutex> guard(g_subclassedWindowsMutex);
+
         subclassedWindows = std::move(g_subclassedWindows);
         g_subclassedWindows.clear();
+    
+        for (HWND hWnd : subclassedWindows)
+            WindhawkUtils::RemoveWindowSubclassFromAnyThread(hWnd, SubclassProc);

```

Comment:
Why did you move the loop inside the locked region? It was outside to avoid deadlocks. `RemoveWindowSubclassFromAnyThread` calls `SendMessage` which waits for the window to respond. Consider the following execution order:
* `g_subclassedWindowsMutex` is locked in this thread
* The UI thread fires `WM_DESTROY` for a window
  * The `WM_DESTROY` subclass handler tries to lock `g_subclassedWindowsMutex`, and waits for it to become available
* While `g_subclassedWindowsMutex` is still locked here, `RemoveWindowSubclassFromAnyThread` is called
  * `RemoveWindowSubclassFromAnyThread` waits for the window to respond
* Deadlock